openff/interchange/components/smirnoff.py

```diff
@@ -136,7 +134,7 @@ def store_matches(
     @classmethod
     def _from_toolkit(
         cls: Type[T],
-        parameter_handler: T_,
+        parameter_handler: PotentialHandler,
```

Comment:
No issues with renaming, the `[A-Z]_` pattern is me trying to follow convention. I'd like to keep the variable in place, though, since it's the easiest way to keep Mypy happy. Alternatively, getting it passing with this approach would work, though I'm not sure how to do that off the top of my head. This is a base class whose derived classes will probably get subclasses of `PotentialHandler` and I think `TypeVar` is one way to get that to work. Open to other solutions - I always feel behind the curve / just barely staying above water with the creative solutions to type annotations.

```suggestion
        parameter_handler: TP,
```

and the corresponding change on L74 (the UI won't let me suggest changes on deleted code ... 🤔 )

Reply:
Oh I thought MyPy would accept subclasses. Can you link me to some examples of that convention? It would help me make the case to fix this bug upstream.

Reply:
Actually I think the problem here might be that there's actually an incorrect type being passed, and that the `bound` argument takes an actual type rather than a string. MyPy's error is:

```
openff/interchange/components/interchange.py:253: error: Argument "parameter_handler" to "_from_toolkit" of "SMIRNOFFPotentialHandler" has incompatible type "List[Any]"; expected "PotentialHandler"  [arg-type]
```

Which AFAIK should not be fixed by having a TypeVar with the PotentialHandler bound. MyPy also [claims](https://mypy.readthedocs.io/en/stable/kinds_of_types.html#class-types) that types should accept their subclasses. I think maybe the argument is incorrectly typed and the TypeVar is masking this because it's incorrectly accepting a string as the Any type? This might be something to leave for a future PR though.

Reply:
Looks like my impression of the convention was wrong ... the docs do not tend to have the trailing underscore. I must have developed the habit reading others' code.

I think you're right ... the electrostatics treatment involves taking a list of handlers and probably other issues as well. I'm fine with the false positive as is, it can be fixed in the future as typing further creeps into things.